src/cluster_legacy.h

```diff
@@ -368,6 +368,7 @@ struct clusterState {
     clusterNode *myself; /* This node */
     uint64_t currentEpoch;
     int state;              /* CLUSTER_OK, CLUSTER_FAIL, ... */
+    int fail_reason;        /* Why the cluster state changes to fail. */
```

Comment:
can we include `fail_reason` in the `CLUSTER INFO` output as well?

Reply:
that is a good idea, i add it to CLUSTER INFO, but it also made it a major decision.